src/openms/source/FORMAT/FileTypes.cpp

```diff
@@ -62,67 +64,67 @@ namespace OpenMS
+    TypeNameBinding(FileTypes::MZQUANTML, "mzq", "mzQuantML file", {}),
+    TypeNameBinding(FileTypes::QCML, "qcml", "quality control file", {}),
+    TypeNameBinding(FileTypes::MZQC, "mzqc", "quality control file in json format", {}),
+    TypeNameBinding(FileTypes::GELML, "gelML", "gelML file", {}),
+    TypeNameBinding(FileTypes::TRAML, "traML", "transition file", {}),
+    TypeNameBinding(FileTypes::MSP, "msp", "NIST spectra library file format", {}),
+    TypeNameBinding(FileTypes::OMSSAXML, "omssaXML", "omssaXML file", {}),
+    TypeNameBinding(FileTypes::MASCOTXML, "mascotXML", "mascotXML file", {}),
+    TypeNameBinding(FileTypes::PNG, "png", "portable network graphics file", {}),
+    TypeNameBinding(FileTypes::XMASS, "fid", "XMass analysis file", {}),
+    TypeNameBinding(FileTypes::TSV, "tsv", "tab-separated file", {}),
+    TypeNameBinding(FileTypes::MZTAB, "mzTab", "mzTab file", {}),
+    TypeNameBinding(FileTypes::PEPLIST, "peplist", "SpecArray file", {}),
+    TypeNameBinding(FileTypes::HARDKLOER, "hardkloer", "hardkloer file", {}),
+    TypeNameBinding(FileTypes::KROENIK, "kroenik", "kroenik file", {}),
+    TypeNameBinding(FileTypes::FASTA, "fasta", "FASTA file", {}),
+    TypeNameBinding(FileTypes::EDTA, "edta", "enhanced dta file", {}),
+    TypeNameBinding(FileTypes::CSV, "csv", "comma-separated values file", {}),
+    TypeNameBinding(FileTypes::TXT, "txt", "generic text file", {}),
+    TypeNameBinding(FileTypes::OBO, "obo", "controlled vocabulary file", {}),
+    TypeNameBinding(FileTypes::HTML, "html", "any HTML file", {}),
+    TypeNameBinding(FileTypes::ANALYSISXML, "analysisXML", "analysisXML file", {}),
+    TypeNameBinding(FileTypes::XSD, "xsd", "XSD schema format", {}),
+    TypeNameBinding(FileTypes::PSQ, "psq", "NCBI binary blast db", {}),
+    TypeNameBinding(FileTypes::MRM, "mrm", "SpectraST MRM list", {}),
+    TypeNameBinding(FileTypes::SQMASS, "sqMass", "SQLite format for mass and chromatograms", {}),
+    TypeNameBinding(FileTypes::PQP, "pqp", "pqp file", {}),
+    TypeNameBinding(FileTypes::MS, "ms", "SIRIUS file", {}),
+    TypeNameBinding(FileTypes::OSW, "osw", "OpenSwath output files", {}),
+    TypeNameBinding(FileTypes::PSMS, "psms", "Percolator tab-delimited output (PSM level)", {}),
+    TypeNameBinding(FileTypes::PIN, "pin", "Percolator tab-delimited input (PSM level)", {}),
+    TypeNameBinding(FileTypes::PARAMXML, "paramXML", "OpenMS internal XML file", {}),
+    TypeNameBinding(FileTypes::SPLIB, "splib", "SpectraST binary spectral library file", {}),
+    TypeNameBinding(FileTypes::NOVOR, "novor", "Novor custom parameter file", {}),
+    TypeNameBinding(FileTypes::XQUESTXML, "xquest.xml", "xquest.xml file", {}),
+    TypeNameBinding(FileTypes::SPECXML, "spec.xml", "spec.xml file", {}),
+    TypeNameBinding(FileTypes::JSON, "json", "JavaScript Object Notation file", {}),
+    TypeNameBinding(FileTypes::RAW, "raw", "(Thermo) Raw data file", {FileTypes::PROVIDES_MS1, FileTypes::PROVIDES_MS2}),
```

Comment:
Readable but not writable

Reply:
ah maybe readable will be implicit?

Reply:
We do have a couple of formats for which we have write support in the handler but not read support. So I don't think it makes sense to make it implicit.

Reply:
Right now there isn't anywhere where we are actually using these attributes to assemble list of filetypes. I think thats a good idea for "Part 2" of this. My idea is to define the features needed when writing the code (EG `FileHandler().LoadExperiment(fname, exp, FileTypesList::typesWithProperties({FileTypes::READABLE, FileTypes::PROVIDESMS1}), log_type_)` in the library, and also populating the TOPP tools allowed types from the typesWithProperties as well.